test/test_csvs.py

```diff
@@ -0,0 +1,291 @@
+# OOIPLACEHOLDER
+#
+# Copyright 2018 Raytheon Co.
+
+import io
+import os
+import unittest
+from nose.plugins.attrib import attr
+
+@attr('UNIT')
+class GeneralUnitTest(unittest.TestCase):
+    # declare the global validation objects
+    _refdes = set()
+    _units = set()
+    _parameters = {}
+    _types = {}
+
+    def _getpath(self, filename):
+        return os.path.join(os.path.dirname(__file__), filename)
+    
+    def setUp(self):
+        # initialize the global objects once
+        if not self._refdes:
+            with io.open(self._getpath('refdes.txt'),mode='rt',encoding='iso-8859-1') as f:
+                for line in f:
+                    self._refdes.add(line.strip())
+        if not self._units:
+            self._units.add('') # empty value is valid
+            with io.open(self._getpath('unit.txt'),mode='rt',encoding='iso-8859-1') as f:
+                for line in f:
+                    self._units.add(line.strip())
+        if not self._parameters:
+            with io.open(self._getpath('parameter.txt'),mode='rt',encoding='iso-8859-1') as f:
+                for line in f:
+                    parts = line.strip().split(':')
+                    self._parameters[parts[0]] = parts[1]
+        if not self._types:
+            with io.open(self._getpath('type.txt'),mode='rt',encoding='iso-8859-1') as f:
+                for line in f:
+                    parts = line.strip().split(':')
+                    self._types[parts[0]] = parts[1]
+
+    def _get_parameters(self):
+        return self._parameters
+
+    def _get_ref_designators(self):
+        return self._refdes
+
+    def _get_types(self):
+        return self._types
+
+    def _get_units(self):
+        return self._units
+
+    def _is_empty_or_numeric(self, value):
+        # if it's empty it's ok
+        if len(value) == 0: return True
+        # any integer or float value will succeed
+        try:
+            float(value)
+        except ValueError:
+            return False
+
+        return True
+
+    def _get_validation_collection(self):
+        return {'ReferenceDesignator' : self._get_ref_designators(),
+                'ParameterID_R'       : self._get_parameters(),
+                'ParameterID_T'       : self._get_parameters(),
+                '_Units'              : self._get_units(),
+                '_DataLevel'          : ('','L0','L1','L2')
+               }
+
+    def _get_validation_positions(self):
+        return {'_Array ID'    : (0,8),
+                '_Platform ID' : (9,14),
+                '_Instrument'  : (18,27)
+               }
+
+    def _validate_line(self, header, flds_to_validate, line):
+        validation_coll = self._get_validation_collection()
+        validation_pos = self._get_validation_positions()
+
+        invalid = []
+        parts = line.split('|')
```

Comment:
Prefer use of established tools (`import csv`) to manual parsing techniques (`line.split('|')`). If the calling function wants to send data to this file, it can just send the line:

```python
for row in csvreader:
    _validate_line(header, fields_to_validate, row)
...
def _validate_line(self, header, fields_to_validate, line):
...
    for index, part in enumerate(row):
```

You save enough lines to buy vowels for `fields_to_validate`. ;)